src/bubo-ht.h

```diff
@@ -166,185 +114,100 @@ class BuboHashSet {
     void get_stats(BuboHashStat* stat) const {
 
         stat->spine_len = table_size_;
-        stat->spine_use = table_curr_use_;
         stat->entries = num_entries_;
-        stat->total_chain_len = 0;
-        stat->collision_slots = 0;
-        stat->max_chain_len = 0;
-        stat->avg_chain_len = 0;
-        stat->dist_1_2 = 0;
-        stat->dist_3_5 = 0;
-        stat->dist_6_9 = 0;
-        stat->dist_10_ = 0;
 
         stat->ht_bytes = table_size_ * sizeof(Entry);
 
         for (uint32_t idx = 0; idx < table_size_; idx++) {
-            Entry* p = &table_[idx];
-            uint64_t chain_len = 0;
-
-            p = p->next_;
-            if (p) stat->collision_slots ++;
-
-            while (p) {
-                stat->ht_bytes += sizeof(Entry);
-                chain_len ++;
-                p = p->next_;
-            }
-
-            if (chain_len >=1 && chain_len <=2) stat->dist_1_2 ++;
-            else if (chain_len >=3 && chain_len <=5) stat->dist_3_5 ++;
-            else if (chain_len >=6 && chain_len <=9) stat->dist_6_9 ++;
-            else if (chain_len >=10) stat->dist_10_ ++;
-
-            stat->total_chain_len += chain_len;
-            if (chain_len > stat->max_chain_len) stat->max_chain_len = chain_len;
-        }
-
-        if (stat->collision_slots > 0) {
-            stat->avg_chain_len = (double) stat->total_chain_len/ (double)stat->collision_slots;
-        } else {
-            assert(stat->total_chain_len == 0);
+            stat->ht_bytes += sizeof(Entry);
         }
 
         uint64_t allocated_bytes = 0, used_bytes = 0;
         blob_store_->stats(&allocated_bytes, &used_bytes);
         stat->blob_allocated_bytes = allocated_bytes;
         stat->blob_used_bytes = used_bytes;
+        stat->bit_set_bytes = bit_set_.size() / 8;
 
-        stat->bytes = stat->ht_bytes + stat->blob_allocated_bytes;
+        stat->bytes = stat->ht_bytes + stat->blob_allocated_bytes + stat->bit_set_bytes;
     }
 
 protected:
     struct Entry {
         const BYTE* val_;
```

Comment:
There's no need to have an Entry struct any more -- The table should just be an array of `BYTE*`